src/dbnode/persist/fs/types.go

```diff
@@ -489,3 +490,45 @@ type BlockRetrieverOptions interface {
 	// IdentifierPool returns the identifierPool
 	IdentifierPool() ident.Pool
 }
+
+// ForEachRemainingFn is the function that is run on each of the remaining
+// series of the merge target that did not intersect with the fileset.
+type ForEachRemainingFn func(seriesID ident.ID, tags ident.Tags) error
+
+// MergeWith is an interface that the fs merger uses to merge data with.
+type MergeWith interface {
+	// Read returns the data for the given block start and series ID, whether
+	// any data was found, and the error encountered (if any).
+	Read(
+		seriesID ident.ID,
+		blockStart xtime.UnixNano,
+		nsCtx namespace.Context,
+	) ([]xio.BlockReader, bool, error)
+
+	// ForEachRemaining loops through each seriesID/blockStart combination that
+	// was not already handled by a call to Read().
+	ForEachRemaining(blockStart xtime.UnixNano, fn ForEachRemainingFn) error
+}
+
+// Merger is in charge of merging filesets with some target MergeWith interface.
+type Merger interface {
+	// Merge merges the specified fileset file with a merge target.
+	Merge(
+		fileID FileSetFileIdentifier,
+		mergeWith MergeWith,
+		flushPreparer persist.FlushPreparer,
+		nsCtx namespace.Context,
+	) error
+}
+
+// NewMergerFn is the function to call to get a new Merger. Mostly used as a
```

Comment:
hm if it's only used for testing, does it need to be exported?

Reply:
That comment is confusing - I removed it. What I meant is that I needed the ability to pass in the merger constructor cold flush code in `shard.go` (a different package) so that it is testable.